### ISSUE TITLE
Collision-related crashes/falling through floor since 3.0.3-rc3

### ISSUE THREAD

**Issue by eiszfuchs** [godotengine/godot#19348]
**Godot version:**
Crash in 3.0.2 until 3.0.3-rc2, collision glitches since RC3

**OS/device including version:**
Windows 10
OpenGL ES 3.0 Renderer: GeForce GTX 1070/PCIe/SSE2

Wasn't able to find any related crash reports, so I'm opening a new one. If I'm incorrect, please feel free to mark this as duplicate. It's quite hard to find related issues if the only thing you have is the stacktrace below.

**Issue description:**
I was playing my game prototype today to verify all levels are doable. Using 3.0.3-rc3, I was hoping to test whether the engine works okay with the recent bug fixes, too. I noticed that the player would often fall through the floor, so I switched back to 3.0.3-rc2 and everything was fine, until this one last level I couldn't finish because it kept crashing at the same point every time. So I tried the 3.0.3-rc3 again and it wasn't crashing, but the player was falling through the floor again at the same spot I was usually experiencing the crash.
It crashes on the latest stable, too.

Terminal leaves me with this:

```
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] <couldn't map PC to fn name>
[1] <couldn't map PC to fn name>
[2] <couldn't map PC to fn name>
[3] <couldn't map PC to fn name>
[4] <couldn't map PC to fn name>
[5] <couldn't map PC to fn name>
[6] <couldn't map PC to fn name>
[7] <couldn't map PC to fn name>
[8] <couldn't map PC to fn name>
[9] <couldn't map PC to fn name>
[10] <couldn't map PC to fn name>
[11] <couldn't map PC to fn name>
[12] <couldn't map PC to fn name>
[13] <couldn't map PC to fn name>
[14] <couldn't map PC to fn name>
[15] BaseThreadInitThunk
-- END OF BACKTRACE --
```

Of course, this isn't too helpful. Is there a build I can try to get a proper stacktrace? I couldn't really narrow down the problem. It's your typical physics-based "roll a ball through a level" game, collecting coins opens a box in which you have to drop. Nothing fancy. The door of the box is animated using Tweens. I don't experience falling through the floor until the goal is opened, so that might be connected.

**Comment by eiszfuchs** [comment]
Ah. Okay. I was able to narrow it down:
This is my highly professional goal-script:

```gd
extends Spatial

const player_class = preload("res://Scripts/Main/Sub/Player.gd")


const level_type = "tile/goal/default"
var tile_id = -1

var tween
var doors = []


func _ready():
	tween = get_node("Tween")

	doors = [
			get_node("Doors/StaticBody1"),
			get_node("Doors/StaticBody2"),
			get_node("Doors/StaticBody3"),
			get_node("Doors/StaticBody4"),
		]

	get_node("Doors").show()

func open_door():
	for door in doors:
		# Fixes possible memory leak,
		# https://github.com/godotengine/godot/issues/19297
		tween.remove(door, "scale")

		tween.interpolate_property(door, "scale", door.scale, Vector3(0.0, 1.0, 0.0), 1.5, Tween.TRANS_LINEAR, Tween.EASE_IN_OUT)

	tween.start()

func close_door():
	tween.remove_all()

	for door in doors:
		door.scale = Vector3(1.0, 1.0, 1.0)

func finish():
	Global.emit_signal("goal_reached")

func _on_GoalArea_body_entered(body):
	if !body is player_class:
		return

	finish()
```

I fixed the falling through the floor by changing:

    tween.interpolate_property(door, "scale", door.scale, Vector3(0.0, 1.0, 0.0), 1.5, Tween.TRANS_LINEAR, Tween.EASE_IN_OUT)

into:

    tween.interpolate_property(door, "scale", door.scale, Vector3(0.01, 1.0, 0.01), 1.5, Tween.TRANS_LINEAR, Tween.EASE_IN_OUT)

Does that make any sense to you?

**Comment by Zireael07** [comment]
Why would you tween a scale's component to 0.0?

**Comment by eiszfuchs** [comment]
It makes parts of my goal disappear by that.  Why should `0.0` for scale be a problem?

**Comment by akien-mga** [comment]
There seem to be two different issues in this report: one crash that happens on all versions, due to using a scale of 0 (which should not be allowed, it likely leads to division by 0), and collision issues only in 3.0.3-rc3 (cc @hpvb).

**Comment by Zireael07** [comment]
I'm not sure, to me it reads as the person had both issues because of 0

@eiszfuchs are you experiencing the falling through after fixing the tween?

**Comment by eiszfuchs** [comment]
Uh, setting the scale to `0.01` fixes the falling through the floor, yes. If you can use it, I can try testing more cases with more versions when I've got the time.

**Comment by hpvb** [comment]
I think this was an issue with msvc and LTO. I suspect that this has been fixed in later releases. Can someone please check to see if this is still a problem?

**Comment by AndreaCatania** [comment]
What's about this issue?

**Comment by eiszfuchs** [comment]
I'll try to test this once again, haven't had the time to test or develop lately, sorry!Am 22.09.2018 11:08 vorm. schrieb Andrea Catania <notifications@github.com>:What's about this issue?

**Comment by eiszfuchs** [comment]
Just tested it.
Set my "door closed" scale to `0.0` again - the falling through the floor is still a problem in `3.0.6`.

Sorry it took so long!

**Comment by akien-mga** [comment]
Setting scale to 0 is not supported, it breaks the physics system.
It should not be crashing though, but looking at https://github.com/godotengine/godot/issues/19348#issuecomment-432282807 it doesn't seem to be (anymore), so I'll close.
